pkg/cloud/amazon/aws_kms.go

```diff
@@ -51,6 +55,20 @@ type kmsURIParams struct {
 	auth                  string
 	roleProvider          roleProvider
 	delegateRoleProviders []roleProvider
+	verbose               bool
+}
+
+var reuseKMSSession = settings.RegisterBoolSetting(
+	settings.TenantWritable,
+	"cloudstorage.aws.reuse_kms_session.enabled",
+	"persist the last opened AWS KMS session and reuse it when opening a new session with the same arguments",
+	true,
```

Comment:
Should we make this a metamorphic value so that it tests the branch where there is no cached client.

Reply:
Good call, changed to metamorphic bool